src/glib/base/ds.hpp

```diff
@@ -381,9 +381,9 @@ void TVec<TVal, TSizeTy>::GetSubValV(const TSizeTy& _BValN, const TSizeTy& _EVal
   const TSizeTy BValN=TInt::GetInRng(_BValN, 0, Len()-1);
   const TSizeTy EValN=TInt::GetInRng(_EValN, 0, Len()-1);
   const TSizeTy SubVals=TInt::GetMx(0, EValN-BValN+1);
-  SubValV.Gen(SubVals, 0);
-  for (TSizeTy ValN=BValN; ValN<=EValN; ValN++){
-    SubValV.Add(GetVal(ValN));}
+  SubValV.Resize(SubVals);
+  memcpy(SubValV.ValT, ValT + BValN, SubVals * sizeof(TVal));
```

Comment:
How does this work for vector with elements that have more complex constructors and/or destructors? My guess is this is problematic for stuff like `TStr` where `memcpy` would copy the `char*` pointer, resulting in two `TStr` objects owning same `char*` pointer.